pkg/utils/gardener/namespacedcloudprofile.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package gardener
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/api"
+	"github.com/gardener/gardener/pkg/apis/core"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+)
+
+// GetCloudProfile gets determine whether a given CloudProfile name is a NamespacedCloudProfile or a CloudProfile and returns the appropriate object
+func GetCloudProfile(ctx context.Context, client pkgclient.Client, cloudProfileReference *gardencorev1beta1.CloudProfileReference, namespace string) (*gardencorev1beta1.CloudProfile, error) {
+	switch cloudProfileReference.Kind {
+	case constants.CloudProfileReferenceKindCloudProfile:
+		cloudProfile := &gardencorev1beta1.CloudProfile{}
+		err := client.Get(ctx, pkgclient.ObjectKey{Name: cloudProfileReference.Name}, cloudProfile)
+		if err == nil {
+			return cloudProfile, nil
+		}
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+	case constants.CloudProfileReferenceKindNamespacedCloudProfile:
+		namespacedCloudProfile := &gardencorev1beta1.NamespacedCloudProfile{}
+		err := client.Get(ctx, pkgclient.ObjectKey{Name: cloudProfileReference.Name, Namespace: namespace}, namespacedCloudProfile)
+		if err == nil {
+			return &gardencorev1beta1.CloudProfile{Spec: namespacedCloudProfile.Status.CloudProfileSpec}, nil
+		}
```

Comment:
```suggestion
		if err := client.Get(ctx, pkgclient.ObjectKey{Name: cloudProfileReference.Name, Namespace: namespace}, namespacedCloudProfile); err != nil {
			return nil, err
		}
		return &gardencorev1beta1.CloudProfile{Spec: namespacedCloudProfile.Status.CloudProfileSpec}, nil
```

Usually, if possible, we check for `err != nil` in-line and exit early.

Reply:
changed again...